renku/cli/init.py

```diff
@@ -95,97 +98,191 @@ def store_directory(ctx, param, value):
+    show_default=True,
+    callback=validate_name,
+    help='Project name.',
 )
-@click.option('--name', callback=validate_name)
-@click.option('--force', is_flag=True, help='Override project templates.')
+@click.option('--template', help='Provide the name of the template to use.')
+@click.option(
+    '--template-source', help='Provide the templates repository url or path.'
+)
+@click.option(
+    '--template-ref',
+    default='master',
+    help='Specify the reference to checkout on remote template repository.',
+)
+@click.option('--description', help='Describe your project.')
+@click.option(
+    '--print-manifest', is_flag=True, help='Print templates manifest only.'
+)
+@click.option('--force', is_flag=True, help='Override target path.')
 @option_use_external_storage
 @pass_local_client
 @click.pass_context
-def init(ctx, client, directory, name, force, use_external_storage):
-    """Initialize a project."""
+def init(
+    ctx, client, use_external_storage, path, name, template, template_source,
+    template_ref, description, print_manifest, force
+):
+    """Initialize a project in PATH. Default is current path."""
+    # preparation
     if not client.use_external_storage:
         use_external_storage = False
 
     ctx.obj = client = attr.evolve(
-        client,
-        path=directory,
-        use_external_storage=use_external_storage,
+        client, path=path, use_external_storage=use_external_storage
     )
 
-    msg = 'Initialized empty project in {path}'
-    branch_name = None
-
-    stack = contextlib.ExitStack()
-
-    if force and client.repo:
-        msg = 'Initialized project in {path} (branch {branch_name})'
-        merge_args = ['--no-ff', '-s', 'recursive', '-X', 'ours']
-
+    def print_done():
+        click.echo(' DONE ', nl=False)
+        click.echo(click.style('\u2713', fg='green'))
+
+    # verify path status
+    if len(list(client.path.glob('**/*'))) > 0:
+        if not force:
+            error = FileExistsError(
+                'Folder "{0}" is not empty. Please '
+                'add --force flag to transform it into a Renku repository.'.
+                format(str(path))
+            )
+            raise click.UsageError(error)
+        else:
+            try:
+                commit = client.find_previous_commit(
+                    str(client.renku_metadata_path),
+                )
+                branch_name = 'pre_renku_init_' + str(commit)[:8]
+                with client.worktree(
+                    branch_name=branch_name,
+                    commit=commit,
+                    merge_args=['--no-ff', '-s', 'recursive', '-X', 'ours']
+                ):
+                    click.echo(
+                        'Saving current data in branch {0}'.
+                        format(branch_name)
+                    )
+            except AttributeError:
+                click.echo('Warning! Overwriting non-empty folder.')
+            except GitCommandError as e:
```

Comment:
Instead of catching it here, we should probably wrap it inside core module and then reraise renku exception.